File: content/snippets/cpp/interview_lcci_10_11_peaks_and_valleys.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

void wiggleSort(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    for (int i = 0; i + 1 < static_cast<int>(nums.size()); i += 2) {
        swap(nums[i], nums[i + 1]);
    }
}

int main() {
    vector<int> nums = {5, 3, 1, 2, 3};
    wiggleSort(nums);
    cout << "[";
    for (int i = 0; i < static_cast<int>(nums.size()); ++i) {
        if (i) cout << ", ";
        cout << nums[i];
    }
    cout << "]" << endl;
    return 0;
}

File: content/snippets/typescript/interview_lcci_10_11_peaks_and_valleys.ts
function wiggleSort(nums: number[]): void {
    nums.sort((a, b) => a - b);
    for (let index = 0; index + 1 < nums.length; index += 2) {
        [nums[index], nums[index + 1]] = [nums[index + 1], nums[index]];
    }
}

const demoValues = [5, 3, 1, 2, 3];
wiggleSort(demoValues);
console.log(demoValues);
